raijin_scripts/execute_sync/run

```diff
@@ -39,83 +37,59 @@ while [[ "$#" -gt 0 ]]; do
+START_YEAR=$(echo "$YEAR" | cut -f 1 -d '-')
+END_YEAR=$(echo "$YEAR" | cut -f 2 -d '-')
+
 # Override echo command to prepend timestamp before echo message
 echo() {
     command echo "$(date '+%F_%T.%N')" "$@"
 }
 
-WORKDIR=/g/data/v10/work/sync/"${PRODUCT}"/$(date '+%FT%H%M')
-SYNC_CACHE="${WORKDIR}"/cache/
-SUBMISSION_LOG="$WORKDIR"/sync-submission-"${PRODUCT}"-$(date '+%F-%T').log
-JOB_NAME="${YEAR}_${PRODUCT}"
-count=0
-dbhostname='agdcdev-db.nci.org.au'
-dbport='6432'
-dbname='datacube'
-
 mkdir -p "${SYNC_CACHE}"
-echo "Start time: " "$(date '+%F-%T')" > "$SUBMISSION_LOG"
 
 module use /g/data/v10/public/modules/modulefiles
 module load "${MODULE}"
-while read -r LINE; do
-  if [[ "$count" -eq 1 ]]
-  then
-      dbhostname="$(cut -d' ' -f2 <<<"$LINE")"
-      count="$((count+1))"
-  elif [[ "$count" -eq 2 ]]
-  then
-      dbport="$(cut -d' ' -f2 <<<"$LINE")"
-      count="$((count+1))"
-  elif [[ "$count" -eq 3 ]]
-  then
-      dbname="$(cut -d' ' -f2 <<<"$LINE")"
-      count="$((count+1))"
-  fi
-
-  if [[ "${STAGE}" == "dev"  && "$LINE" == "[dea-dev]" ]]; then
-      count="$((count+1))"
-  fi
-
-  if [[ "${STAGE}" == "prod"  && "$LINE" == "[datacube]" ]]; then
-      count="$((count+1))"
-  fi
-done  < "$DATACUBE_CONFIG_PATH"
 
 TRASH_ARCHIVED=''
 if [ "$TRASH_ARC" == yes ]; then
    TRASH_ARCHIVED='--trash-archived'
 fi
 
+for (( year = "$START_YEAR"; year <= "$END_YEAR"; ++year )); do
+    PATHS_TO_PROCESS+=("$BASE_PATH$year$SUFFIX_PATH")
+done
+
 cd "${WORKDIR}"
 
-nohup "$SHELL" > "$SUBMISSION_LOG" 2>&1 << EOF &
-echo "Logging job: ${JOB_NAME} into: ${SUBMISSION_LOG}"
-echo ""
-echo Loading module "${MODULE}"
-echo ""
+echo "Start time: $(date '+%F-%T')
+
+Logging job: ${JOB_NAME} into: ${SUBMISSION_LOG}
+Loading module ${MODULE}
 
 # Check if we can connect to the database
-datacube -vv system check
+$(datacube -vv system check)
+" > "$SUBMISSION_LOG"
 
-# Read agdc datasets from the database before sync process
-echo ""
-echo "**********************************************************************"
-echo "Read previous agdc_dataset product names and count before sync process"
-echo "Connected to the database host name: ${dbhostname}"
-echo "Connected to the database port number: ${dbport}"
-echo "Connected to the database name: ${dbname}"
-psql -h ${dbhostname} -p ${dbport} -d ${dbname} \
-  -c 'select name, count(*) FROM agdc.dataset a, agdc.dataset_type b where a.dataset_type_ref = b.id group by b.name'
-echo "**********************************************************************"
+for syncpath in "${PATHS_TO_PROCESS[@]}"
+do
+  nohup "$SHELL" >> "$SUBMISSION_LOG" 2>&1 << EOF &
+  ##################################################################################################
+  # Run dea-sync process
+  ##################################################################################################
+  echo ""
+  echo "qsub -V -N dea-sync -q ${QUEUE} -W umask=33 -l wd,walltime=10:00:00,mem=25GB,ncpus=16 -m ae \
```

Comment:
Instead of _copy/paste + echo_, there's a built in shell command to echo commands before running them which would work better here.

`set -x` to enable and `set +x` to disable.

See [this StackOverflow answer](https://stackoverflow.com/a/2853811/119603) for more details.

We could probably enable it at the top of the shell scripts, or, could enable it just for particular commands we want logged.